base/src/types.rs

```diff
@@ -193,12 +190,6 @@ impl<Id, T> Type<Id, T>
     pub fn kind(&self) -> RcKind {
         use self::Type::*;
         match *self {
-            App(ref arg, _) => {
-                match *arg.kind() {
-                    Kind::Function(_, ref ret) => ret.clone(),
-                    _ => panic!("Expected function kind"),
-                }
-            }
```

Comment:
Definitely fishy though I suspect I only use that method for debugging purposes (hopefully)